File: 2021/03/1.pl
#!/usr/bin/perl

my @sum;
my @numlines;

while(<>)
{
    my @line = split (undef,$_);
    for my $i (0 .. $#line-1)
    {
	$sum[$i]+=$line[$i];
    }
	$numlines++;
}

my $gamma = 0;
my $epsilon = 0;

for my $i (0 .. $#sum)
{
    my $bitc=$sum[$i];
    print "$i $sum[$i] $bitc\n";
    if($bitc > ($numlines / 2))
    {
	$epsilon*=2;
	$gamma*=2;
	print "1";
	$epsilon++;
    }
    else
    {
	$epsilon*=2;
	$gamma*=2;
	print "0";
	$gamma++;
    }
    print "\n gamma $gamma epsilon $epsilon\n";
}

my $power=$gamma*$epsilon;
print "\n gamma $gamma epsilon $epsilon power $power\n";

File: 2022/04/1.py
#!/usr/bin/python3

p1=0
p2=0

with open('input') as f:
    for line in f:
        l,r=line.strip().split(",")
        l1,r1=l.split("-")
        l2,r2=r.split("-")
        print(l1,r1,l2,r2)
        if (int(l1)<=int(l2) and int(r1)>=int(r2)) or (int(l2)<=int(l1)) and (int(r2)>=int(r1)):
            print("over")
            p1+=1
        if (int(l1)>=int(l2) and int(l1)<=int(r2)) or (int(l2)>=int(l1)) and (int(l2)<=int(r1)):
            print("sover")
            p2+=1




print("p1:",p1)
print("p2:",p2)
